projects/cardiovascular-patient-app/src/app/last-report-bp/last-report-bp.component.spec.ts

```diff
@@ -0,0 +1,170 @@
+      providers: [
+        { provide: DataLayerManagerService, useValue: layerManager }
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(LastReportBPComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should show correct message for systolic and diastolic of last reported prior BP with multiple measurements', async () => {
+    const layers: ManagedDataLayer[] = [
+      {
+        id: '1',
+        name: 'Blood Pressure',
+        category: ['C', 'D'],
+        datasets: [
+          {
+            data: [
+              { x: 1609175027000, y: 120 },
+              { x: 1609175027000, y: 122 },
+            ],
+            label: "Diastolic Blood Pressure"
+          },
+          {
+            data: [
+              { x: 1676536294000, y: 80 },
+              { x: 1676536294000, y: 78 },
+            ],
+            label: "Systolic Blood Pressure"
+          },
+        ],
+        scale: { id: '1' },
+      },
+    ];
+    layerManager.allLayers$.next(layers);
+
+    const cardHarness = await TestbedHarnessEnvironment.harnessForFixture(fixture, MatCardHarness);
+    const contentText = await cardHarness.getText();
+
+    const expectedOutput: LastReportedBPdata = {
+      systolic: { date: new Date(1676536294000).toUTCString(), value: 78 },
+      diastolic: { date: new Date(1676536294000).toUTCString(), value: 122 },
+    };
+    expect(contentText).toContain(`Last BP reported was ${component?.lastReportedBPdata?.systolic.value }/${ component?.lastReportedBPdata?.diastolic.value } on ${component.formatDate(expectedOutput.systolic.date)} at ${component.formatTime(expectedOutput.systolic.date)}`);
+  });
+
+  it('should show correct text for systolic and diastolic values of last reported prior BP if the Blood Pressure layer has no datasets', async () => {
+    const layers: ManagedDataLayer[] = [
+      {
+        id: '1',
+        name: 'Medications',
+        category: ['C', 'D'],
+        datasets: [
+          {
+            data: [
+              { x: 1609175027000, y: 120 },
+              { x: 1609175027000, y: 122 },
+            ],
+          },
+          {
+            data: [
+              { x: 1676536294000, y: 80 },
+              { x: 1676536294000, y: 78 },
+            ],
+          },
+        ],
+        scale: { id: '1' },
+      },
+    ];
+    layerManager.allLayers$.next(layers);
+    const cardHarness = await TestbedHarnessEnvironment.harnessForFixture(fixture, MatCardHarness);
+    const contentText = await cardHarness.getText();
+    expect(contentText).toContain('There is no last reported prior BP for Patient');
+  });
+
+  it('should return the correct systolic and diastolic most recent blood pressure values for a patient with multiple measurements', async () => {
+    const layers: ManagedDataLayer[] = [
+      {
```

Comment:
It looks like these two tests that you just added are duplicating the ones above. There's no point in checking intermediate values if you already have a test that checks the DOM.

Reply:
Ok @semicolin, i will remove that  intermediate values and check the test cases